lib/jnpr/junos/facts/get_software_information.py

```diff
@@ -84,11 +93,22 @@ def get_facts(device):
         if re_version is None:
             # For < 15.1, get version from the "junos" package.
             try:
-                re_pkg_info = re_sw_info.xpath(
+                junos_pkg_info = re_sw_info.xpath(
                     './package-information[name="junos"]/comment'
-                )[0].text
+                )
+                if len(junos_pkg_info):
+                    re_pkg_info = junos_pkg_info[0].text
+                else:
+                    # check JDM variants
+                    jdm_pkg_info = re_sw_info.xpath(
+                        './package-information[starts-with(name, "Junos")]/comment'
+                    )
+                    if len(jdm_pkg_info):
+                        re_pkg_info = jdm_pkg_info[0].text
+
                 re_version = re.findall(r'\[(.*)\]', re_pkg_info)[0]
-            except:
+
+            except Exception:
```

Comment:
Thanks for cleaning this up. `except Exception:` is definitely better than the `except:` I had.